spec/formatter_spec.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const {format} = require('../src/fxp');
+
+describe('XML formatter', () => {
+  it('should format simple document', () => {
+    const result = format('<data><item>1</item></data>');
+    expect(result).toBe(`
+<data>
+  <item>
+    1
+  </item>
```

Comment:
I find this choice a bit surprising. Personally, I'd prefer the text to be on the same line as the opening tag.

Here is an example showing how `xmllint` handles formatting:

```
$ echo "<data><item>1</item></data>" | xmllint --format -
<?xml version="1.0"?>
<data>
  <item>1</item>
</data>
```

Anyhow, I think changing the formatting rules is out of scope of this pull request. If you are ok with changing the formatter to make it closer to `xmllint`, then I am happy to open another pull request.

Reply:
I see this has been discussed in https://github.com/NaturalIntelligence/fast-xml-parser/discussions/401 with the following conclusion:

> It's difficult to decide when opening and closing tags should be in-lined text data. However, we can provide a configuration for this.

Can you @amitguptagwl please show me some pointers on how & where to start implementing such configuration?